tokio-util/src/codec/encoder.rs

```diff
@@ -18,5 +18,5 @@ pub trait Encoder {
     /// This method will encode `item` into the byte buffer provided by `dst`.
     /// The `dst` provided is an internal buffer of the `Framed` instance and
     /// will be written out when possible.
-    fn encode(&mut self, item: Self::Item, dst: &mut BytesMut) -> Result<(), Self::Error>;
+    fn encode(&mut self, item: &Item, dst: &mut BytesMut) -> Result<(), Self::Error>;
```

Comment:
Didn't think about the lifetime issues :/ I am not able to decide on this obviously but `&Item` with `Item` as an argument seems like the way to go 👍